specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -363,7 +365,7 @@ export function QueryBuilder({
         }}
       >
         <div className="flex snap-start flex-col gap-4 overflow-y-auto">
-          {state.showMappingView && (
+          {state.showMappingView && showEmbeddedMappingView ? (
```

Comment:
why add a new prop when you could already reuse existing one? `state.showMappingView`

Reply:
not sure what you mean, since I had a toggle btn in the parent I needed to pass down the boolean. showMappingView doesn't come from the parent of QueryBuilder.

Reply:
ah, I overlooked that.

Still, you can refactor the code (and simplify it in the process) to fix this

take a look at where the initial value for `state.showMappingView` is coming from and what happens when user tries to call (what does the action handler do) and think of how you can improve that

<details><summary>Solution</summary>
<p>

Remove `showMappingView` from state and use `useCachedState` directly.
It is not using it at the moment because query builder was developed when I (mistakingly) believed that having a single big state instead of several smaller is best.

Then, you can use `useCachedState` outside of this component too, and any changes to the state will be synced between all usages

</p>
</details>